membership-attribute-service/app/services/zuora/payment/PaymentService.scala

```diff
@@ -84,17 +83,17 @@ class PaymentService(zuoraService: ZuoraSoapService, planMap: Map[ProductRatePla
     }
 
   private def getNextBill(subId: Id, account: Account, numberOfBills: Int)(implicit logPrefix: LogPrefix): Future[Option[Bill]] =
-    zuoraService.previewInvoices(subId, numberOfBills).map { previewInvoiceItems =>
-      val maybeBillingSchedule = BillingSchedule.fromPreviewInvoiceItems(planMap, previewInvoiceItems)
-      maybeBillingSchedule.flatMap { billingSched =>
-        billingSched
-          .withCreditBalanceApplied(account.creditBalance)
-          .invoices
-          .list
-          .find(_.amount > 0)
-          .toOption
-      }
-    }
+    for {
+      previewInvoiceItems <- zuoraService.previewInvoices(subId, numberOfBills)
+    } yield for {
+      billingSched <- BillingSchedule.fromPreviewInvoiceItems(previewInvoiceItems)
```

Comment:
only change is remove planMap from the parameters on this line. 
Also converted the nested code into for comprehensions